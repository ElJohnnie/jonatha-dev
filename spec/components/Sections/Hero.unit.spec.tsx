import { NextIntlClientProvider } from 'next-intl';
import HomeView from '@/app/[locale]/home/view/home.view';
import { render } from '@testing-library/react';
import React from 'react';

const heroMockup = {
  title: 'any title',
  description: 'any description',
};

const Wrapper = ({ children }: { children: React.ReactNode }) => (
  <NextIntlClientProvider locale="en" messages={{}}>
    {children}
  </NextIntlClientProvider>
);

describe('should render correctly Hero component', () => {
  beforeAll(() => {
    global.IntersectionObserver = class {
      constructor(
        _callback: IntersectionObserverCallback,
        _options?: IntersectionObserverInit
      ) {}
      observe() {}
      unobserve() {}
      disconnect() {}
      takeRecords() {
        return [];
      }
      root: Element | null = null;
      rootMargin: string = '';
      thresholds: ReadonlyArray<number> = [];
    };
  });

  it('match snapshot', () => {
    const { container } = render(
      <Wrapper>
        <HomeView {...heroMockup} />
      </Wrapper>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});