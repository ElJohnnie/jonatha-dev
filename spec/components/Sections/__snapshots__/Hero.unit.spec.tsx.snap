// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly Hero component match snapshot 1`] = `
<section
  class="mt-8 h-auto pt-8"
>
  <div
    class="max-w-screen mx-auto grid justify-center px-4 py-12"
  >
    <div
      class="mr-auto mt-[10vh] place-self-center md:col-span-12"
    >
      <h1
        class="mb-4 max-w-2xl text-3xl font-extrabold leading-none tracking-tight md:text-6xl dark:text-white"
      >
        <span
          class="leading-12 block w-full bg-gradient-to-r from-indigo-700 to-fuchsia-500 bg-clip-text py-2 text-transparent lg:inline"
        >
          any title
        </span>
      </h1>
      <div
        class="md:inline-flex"
      >
        <h1
          class="text-1xl bg-gradient-to-r from-slate-100 to-slate-200 to-50% bg-clip-text font-extrabold tracking-tight text-transparent [text-wrap:balance] md:text-4xl"
        >
          any description
           
        </h1>
        <span
          class="text-1xl inline-flex h-[calc(theme(fontSize.base)*theme(lineHeight.tight))] flex-col overflow-hidden font-extrabold text-indigo-500 md:ml-1 md:h-[calc(theme(fontSize.4xl)*theme(lineHeight.tight))] md:text-4xl"
        >
          <ul
            class="block animate-text-slide-4 text-left leading-tight [&_li]:block"
          >
            <li
              aria-hidden="true"
            />
          </ul>
        </span>
      </div>
      <div
        class="mt-4 flex flex-row justify-center"
      >
        <a
          class="align-center group relative inline-flex items-center justify-center overflow-hidden rounded border border-transparent bg-transparent px-12 py-3 text-sm font-bold text-slate-100 hover:text-indigo-500 focus:outline-none focus:ring active:bg-indigo-600 active:text-white"
          href="/about"
        >
          <span
            class="ease absolute left-0 top-0 h-0 w-0 border-t-2 border-indigo-500 transition-all duration-200 group-hover:w-full"
          />
          <span
            class="ease absolute right-0 top-0 h-0 w-0 border-r-2 border-indigo-500 transition-all duration-200 group-hover:h-full"
          />
          <span
            class="ease absolute bottom-0 right-0 h-0 w-0 border-b-2 border-indigo-500 transition-all duration-200 group-hover:w-full"
          />
          <span
            class="ease absolute bottom-0 left-0 h-0 w-0 border-l-2 border-indigo-500 transition-all duration-200 group-hover:h-full"
          />
          <span
            class="flex items-center justify-center align-middle"
          />
        </a>
      </div>
    </div>
  </div>
</section>
`;
